### DISCUSSION TITLE
Moving to a new controller

### DISCUSSION THREAD

**Discussion by majorsl** [zwave-js/zwave-js-ui#3831] [Q&A]
I have a HUSBZB-1 and am thinking of moving to a newer controller.  I am not entirely sure that HUSBZB-1 supports priority routes or not for example (at least I can't find any info on it).

I'm not clear on the process.  I have backups, but not NVM (yet.)  I read the backing up NVM between different model controllers is not a good idea.

Will restoring the other backup with a new stick attached "just work"?  I also read about having the new stick as a secondary controller and then promoting it.  I don't know if that was for this project or some other software.

**Comment by robertsLando** [comment]
The most important thing when changing controllers is the NVM backup as that's the backup of controller memory.

The only one that can aswer your question about compatibility issues between NVM backups is @AlCalzone

**Comment by kpine** [comment]
> Will restoring the other backup with a new stick attached "just work"? I also read about having the new stick as a secondary controller and then promoting it. I don't know if that was for this project or some other software.

Not with the Nortek, unless you happen to have a special one with a newer SDK than most. The very old SDK doesn't support the NVM format that Z-Wave JS requires for migration. You need to perform a rather invasive firmware upgrade process which updates it to supporting SDK version: https://community.hubitat.com/t/guide-nortek-husbzb-1-nvm-backup-restore-and-updating-z-wave-firmware/48012

For the secondary controller method, you have to use the Windows PC Controller software. The process is demonstrated at https://www.youtube.com/watch?v=lktIFOcg7GI, but then you need to take care as the Nortek doesn't support Virtual Nodes and you have to do with the first commenter did:

> I ran into one issue where my old controller did not display the "Add/Remove Virtual" button.  I worked around this by adding up to virtual device 231 on the new controller, then removing all of those except node 231, then when I shifted it, the old controller became node 232.  That then allowed me to add the new controller as node 1 when I shifted back to it.

**Comment by majorsl** [comment]
> > Will restoring the other backup with a new stick attached "just work"? I also read about having the new stick as a secondary controller and then promoting it. I don't know if that was for this project or some other software.
> 
> Not with the Nortek, unless you happen to have a special one with a newer SDK than most. The very old SDK doesn't support the NVM format that Z-Wave JS requires for migration. You need to perform a rather invasive firmware upgrade process which updates it to supporting SDK version: https://community.hubitat.com/t/guide-nortek-husbzb-1-nvm-backup-restore-and-updating-z-wave-firmware/48012
> 
> For the secondary controller method, you have to use the Windows PC Controller software. The process is demonstrated at https://www.youtube.com/watch?v=lktIFOcg7GI, but then you need to take care as the Nortek doesn't support Virtual Nodes and you have to do with the first commenter did:
> 
> > I ran into one issue where my old controller did not display the "Add/Remove Virtual" button.  I worked around this by adding up to virtual device 231 on the new controller, then removing all of those except node 231, then when I shifted it, the old controller became node 232.  That then allowed me to add the new controller as node 1 when I shifted back to it.

Thanks! Wow. What an absolute nightmare this is turning out to be. Just. Wow. ...and even the backups I have now are worthless because if the HUSBZB-1 fails, I can really only restore to an identical obsolete one.

I'm almost better off running a second Z-JS instance in docker and going through the pain of repairing everything when I have time to do a few of the devices at a time.

**Comment by kpine** [comment]
> I'm almost better off running a second Z-JS instance in docker and going through the pain of repairing everything when I have time to do a few of the devices at a time.

How many devices do you have? That's probably what I'd do.

**Comment by majorsl** [comment]
> > I'm almost better off running a second Z-JS instance in docker and going through the pain of repairing everything when I have time to do a few of the devices at a time.
> 
> How many devices do you have? That's probably what I'd do.

82. Any thoughts on what the current "best" modern zwave USB sticks are?

**Comment by AlCalzone** [comment]
According to very recent tests:
1. HomeSeer G8
2. Zooz ZST39 

The Nabu Casa stick is going to take the first spot when it is out, but that can still take a while.

I don't have any other 800 series sticks though, seems a few others came out recently.

**Comment by majorsl** [comment]
> According to very recent tests:
> 
> 1. HomeSeer G8
> 2. Zooz ZST39
> 
> The Nabu Casa stick is going to take the first spot when it is out, but that can still take a while.
> 
> I don't have any other 800 series sticks though, seems a few others came out recently.

Thanks! Other than a zwave door lock (non-plus) that likes to fall off the network because it doesn't realize there are 3 wired devices closer, I'm not in a big hurry.  It might be easier to get the Zwave+ module for that one in the short term anyway.
